vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -11,23 +10,158 @@
+  if (!host) {
+    return;
+  }
+
+  RegisterWndClass(); // creates and register a windows class
+
+  if (!m_hwnd) {
+    auto CompositionHwndHost = winrt::Microsoft::ReactNative::CompositionHwndHost();
+    winrt::Microsoft::ReactNative::ReactViewOptions viewOptions;
+    viewOptions.ComponentName(L"Modal");
+    CompositionHwndHost.ReactViewHost(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::MakeViewHost(host, viewOptions));
+    HINSTANCE hInstance = GetModuleHandle(NULL);
+    winrt::impl::abi<winrt::Microsoft::ReactNative::ICompositionHwndHost>::type *pHost{nullptr};
+    winrt::com_ptr<::IUnknown> spunk;
+    CompositionHwndHost.as(spunk);
+    spunk->AddRef(); // Will be stored in windowData
+
+    // get the root hwnd
+    auto roothwnd = reinterpret_cast<HWND>(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::GetTopLevelWindowId(m_context.Properties().Handle()));
+
+    m_hwnd = CreateWindow(
+        c_modalWindowClassName,
+        L"React-Native Modal",
+        WS_OVERLAPPEDWINDOW,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        MODAL_DEFAULT_WIDTH,
+        MODAL_DEFAULT_HEIGHT,
+        roothwnd, // parent
+        nullptr,
+        hInstance,
+        spunk.get());
+  }
+
+  ShowWindow(m_hwnd, SW_NORMAL);
+  BringWindowToTop(m_hwnd);
+  SetFocus(m_hwnd);
+}
+
+void WindowsModalHostComponentView::HideOnUIThread() noexcept {
+  if (m_hwnd) {
+    ::ShowWindow(m_hwnd, SW_HIDE);
+  }
+}
+
+// Windows Procedure - callback function used for handling all messages (generated by NTUser or manual calls to
+// SendMessage)
+LRESULT CALLBACK LogBoxWndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+  auto data = reinterpret_cast<::IUnknown *>(GetProp(
+      hwnd,
+      CompHostProperty)); // gets data handle from the property list of specified window (ie the window we want to make)
+  winrt::com_ptr<winrt::IUnknown> spunk;
+  winrt::Microsoft::ReactNative::CompositionHwndHost host{nullptr};
+
+  if (data) {
+    winrt::check_hresult(data->QueryInterface(
+        winrt::guid_of<winrt::Microsoft::ReactNative::CompositionHwndHost>(),
+        winrt::put_abi(host))); // look into the data for a CompositionHwndHost and store it in host
+    auto result = static_cast<LRESULT>(host.TranslateMessage(message, wparam, lparam));
+    if (result)
+      return result;
+  }
+
+  switch (message) {
+    case WM_NCCREATE: { // sent before WM_CREATE, lparam should be identical to members of CreateWindowEx
+      auto createStruct = reinterpret_cast<CREATESTRUCT *>(lparam); // CreateStruct
+      auto windowData = static_cast<::IUnknown *>(createStruct->lpCreateParams);
+      SetProp(hwnd, CompHostProperty, reinterpret_cast<::IUnknown *>(windowData)); // adds new properties to window
+      break;
+    }
+    case WM_CREATE: { // recieves after window is created but before visible
+      // host.Initialize((uint64_t)hwnd); // this line currently breaks Modal and LogBox
```

Comment:
Breaks how?

Reply:
This was fixed last week via https://github.com/microsoft/react-native-windows/pull/12725!

Reply:
Fixed for LogBox, not for Modal :( will leave this commented out for now to address in a later PR because this is needed to add RN Components into Modal (which isn't the intent of this PR)